File: SewerProject/Assets/_Scripts/interlaceRoller.js
var scrollSpeed = 0.1;

private var initialColor : Color;

private var flicker = 0.0;


function Start()
{
	initialColor = renderer.material.GetColor("_TintColor");
}

function Update () 
{   
	var offset = Time.time * scrollSpeed;
    renderer.material.SetTextureOffset ("_Interlace", Vector2(0.0, offset));
    
    var scaleY = Mathf.Sin (Time.time) * 0.2;
    renderer.material.SetTextureScale ("_Interlace", Vector2(2.0,scaleY  + 1.6));
    
    if (Random.value >= 0.7)
	{
		flicker = Random.value + 0.2;
		
		light.intensity = flicker;
		
		renderer.material.SetColor ("_TintColor", Color(initialColor.r * flicker, initialColor.g * flicker, initialColor.b * flicker));
	}
	
	audio.pitch = scaleY + 1.0;	
}

File: Melodious Eras/Assets/Shaders/Objects/InterlaceRoller.cs
using UnityEngine;
using System.Collections;

public class InterlaceRoller : MonoBehaviour 
{	
	public float scrollSpeed = .1f;
    public Light _Light;
    public int screenMatIndex = 0;
	public bool flickering = false;
    public Material screenMat;

	private Color initialColor;
	private float flicker = 0.0f;
    Renderer _renderer;
    AudioSource _audio;

    private delegate void RenderEffects(float scaleY);
    private RenderEffects effects;
	
	void Start()
	{
        //initialColor = this.renderer.materials[screenMatIndex].GetColor("_TintColor");
        _renderer = GetComponent<Renderer>();
        _audio = GetComponent<AudioSource>();
        screenMat = _renderer.materials[screenMatIndex];
        initialColor = screenMat.GetColor("_TintColor");

        if (_Light == null && this.GetComponent<Light>() != null)
            _Light = this.GetComponent<Light>();
	}
	
	void Update()
	{
		float offset = Time.time * scrollSpeed;
        screenMat.SetTextureOffset("_Interlace", new Vector2(0.0f, offset));

		float scaleY = Mathf.Sin(Time.time) * 0.2f;
        screenMat.SetTextureScale("_Interlace", new Vector2(2.0f, scaleY + 1.6f));

		if(flickering && Random.value >= .7f)
		{
			flicker = Random.value + 0.2f;
            screenMat.SetColor("_TintColor", new Color(initialColor.r * flicker, initialColor.g * flicker, initialColor.b * flicker));
            if(_Light != null)
                _Light.intensity = flicker;
		}
		if(_audio)
			_audio.pitch = scaleY + 1;
	}
}
